pathfinder/tests/test_cli.py

```diff
@@ -0,0 +1,111 @@
+    runner = CliRunner()
+    with patch.multiple(**patch_args) as mocks:
+        mocks['get_default_registry_and_start_block'].return_value = Mock(), Mock()
+        result = runner.invoke(main, [])
+    assert result.exit_code == 0
+
+
+def test_eth_rpc():
+    """ The `eth-rpc` parameter must reach the `HTTPProvider` """
+    runner = CliRunner()
+    eth_rpc = 'example.com:1234'
+    with patch('pathfinder.cli.HTTPProvider') as provider:
+        runner.invoke(main, ['--eth-rpc', eth_rpc])
+        provider.assert_called_with(eth_rpc)
+
+
+def test_registry_address():
+    """ The `registry_address` parameter must reach the `PathfindingService` """
+    runner = CliRunner()
+    with patch.multiple(**patch_args) as mocks:
+        address = Web3.toChecksumAddress('0x' + '1' * 40)
+        result = runner.invoke(main, ['--registry-address', address])
+        assert result.exit_code == 0
+        assert mocks['PathfindingService'].call_args[1]['registry_address'] == address
+
+    # check validation of address format
+    def fails_on_registry_check(address):
+        result = runner.invoke(main, ['--registry-address', address], catch_exceptions=False)
+        assert result.exit_code != 0
+        assert 'EIP-55' in result.output
+
+    fails_on_registry_check('1' * 40)  # no 0x
+    fails_on_registry_check('0x' + '1' * 41)  # not 40 digits
+    fails_on_registry_check('0x' + '1' * 39)  # not 40 digits
+
+
+def test_start_block():
+    """ The `start_block` parameter must reach the `PathfindingService`
+
+    We also have to pass a registry address, because `start_block` is
+    overwritten with a default when no registry has been specified.
+    """
+    runner = CliRunner()
+    with patch.multiple(**patch_args) as mocks:
+        mocks['get_default_registry_and_start_block'].return_value = Mock(), Mock()
+        start_block = 10
+        address = Web3.toChecksumAddress('0x' + '1'*40)
+        result = runner.invoke(main, ['--registry-address', address, '--start-block', start_block])
+        assert result.exit_code == 0
+        assert mocks['PathfindingService'].call_args[1]['sync_start_block'] == start_block
+
+
+def test_confirmations():
+    """ The `confirmations` parameter must reach the `PathfindingService` """
+    runner = CliRunner()
+    with patch.multiple(**patch_args) as mocks:
+        mocks['get_default_registry_and_start_block'].return_value = Mock(), Mock()
+        confirmations = 77
+        result = runner.invoke(main, ['--confirmations', confirmations], catch_exceptions=False)
+        assert result.exit_code == 0
+        assert mocks['PathfindingService'].call_args[1]['required_confirmations'] == confirmations
+
+
+def test_default_registry(token_network_registry_contract):
+    """ We can fall back to a default registry if none if specified """
+    net_version = 3
+    contracts_version = 'pre_limits'
+    registry_address, block_number = get_default_registry_and_start_block(
```

Comment:
It should be checked that these are valid values and that the block number is bigger than 0